client/src/components/dashboard/deploy/DeployPage.js

```diff
@@ -18,31 +20,39 @@ const DeployPage = ({ teamId }) => {
     dispatch(deployToGHPages(teamId, accessToken));
   };
 
-  useEffect(() => {
-    // github returns a code in the url after user logs in
-    const url = window.location.href;
-    const hasCode = url.includes('?code=');
-    if (hasCode && !retrievedAccessToken && teamId) {
-      const code = url.split('?code=')[1];
-      // we use this code to exchange an access token
-      dispatch(getGHAccessToken(teamId, code));
-    } else if (!retrievedAccessToken) {
-      // we refreshed so we should clear local storage
-      localStorage.clear();
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [dispatch, teamId]);
+  const onSuccessfulLogin = (response) => {
+    const { code } = response;
+    // Now that we have the temporary code, we wish to exchange it for a GitHub
+    // access token. This action will fetch the token and push it to localstorage.
+    dispatch(getGHAccessToken(teamId, code));
+  }
+
+  const onLoginFail = (response) => {
+    // TODO, show a nice error.
+    console.error(response);
+  }
+
+  // useEffect(() => {
+  //   // github returns a code in the url after user logs in
+  //   const url = window.location.href;
+  //   const hasCode = url.includes('?code=');
+  //   if (hasCode && !retrievedAccessToken && teamId) {
+  //     const code = url.split('?code=')[1];
+  //     // we use this code to exchange an access token
+  //     dispatch(getGHAccessToken(teamId, code));
+  //   } else if (!retrievedAccessToken) {
+  //     // we refreshed so we should clear local storage
+  //     localStorage.clear();
+  //   }
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [dispatch, teamId]);
 
   const GitHubLoginButton = (
-    <Button
-      className="float-right"
-      variant="outline-primary"
-      href={githubLoginUrl}
-      disabled={retrievedAccessToken}
-    >
+    <GitHubLogin className="float-right" clientId={githubClientId} scope={scope}
+                 onSuccess={onSuccessfulLogin} onFailure={onLoginFail}
```

Comment:
Is the reason why you decided to use this GH login component over the previous one is because it has support for checking the login result? That is neat that we don't have to rely on useEffect for this.

Reply:
Yes! I found it much cleaner, but the downside is that the button does not look the best. I'll leave styling up to someone else.

Reply:
I have a ticket on winter's board to refactor the deploy page anyway, so I'll put that in there for someone to handle it.